### ISSUE TITLE
Duplicate memory affinity domain strict check

### ISSUE THREAD

**Issue by ctfhacker** [gamozolabs/chocolate_milk#10]
It's possible to have multiple memory affinities on the same domain. Relaxing this check causes the system to work just fine.

```index c92c4c1..5f0b195 100644
--- a/kernel/src/acpi.rs
+++ b/kernel/src/acpi.rs
@@ -454,8 +454,8 @@ unsafe fn parse_srat(ptr: PhysAddr) ->                // Log the affinity record
                if (flags & FLAGS_ENABLED) != 0 {
                    {+//+} assert!(memory_affinities.insert(domain, (base, [-size))-]
[-                            .is_none(),-]{+size)).is_none(),+} "Duplicate memory affinity domain");
                    {+memory_affinities.insert(domain, (base, size));+}
                }
            }```

Tested on PowerEdge T640 with Intel Xeon SIlver 4110

**Comment by gamozolabs** [comment]
I've seen this as well a few years back on some machines I used. The fix isn't exactly trivial so it might take me some time to get to. Without a machine to test on it'll be relatively hard to make sure I fixed correctly. I'll effectively need to support multiple memory ranges per NUMA node, otherwise I'd miss some ranges. This means anywhere I store these ranges (and check them during allocation), I need to support sets of ranges.
